server/handler/base.go

```diff
@@ -391,6 +402,73 @@ func selectionToReviewersRequest(s reviewer.Selection) github.ReviewersRequest {
 	return req
 }
 
+func (b *Base) dedupDiscardedReviews(discardedReviews []*common.DiscardedReview) []*common.DiscardedReview {
```

Comment:
Seeing as a `DiscardedReview` is an ID and a reason, I wonder if we could avoid the de-duplication function by passing around a `map[id]reason` instead of a slice.

Even if not, I recommend implementing this function by building up a map instead of using the dual loop. I don't think the caller cares whether it iterates over a slice or a map.